File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/sales/sales-reason/sales-reason-lookup-model.ts

export interface ISalesReasonLookupModel {
  salesReasonID: number;
  name: string;
  reasonType: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class SalesReasonLookupModel implements ISalesReasonLookupModel {
  salesReasonID: number;
  name: string;
  reasonType: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<SalesReasonLookupModel>) {
    if (!!init) {
      Object.assign<SalesReasonLookupModel, Partial<SalesReasonLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Sales/SalesReason/Commands/Queries/GetSalesReasons/SalesReasonLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Sales;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;


namespace AdventureWorks.Application.DataEngine.Sales.SalesReason.Queries.GetSalesReasons
{
    public partial class SalesReasonLookupModel : IHaveCustomMapping
    {
        public int SalesReasonID { get; set; }
        public string Name { get; set; }
        public string ReasonType { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Sales.SalesReason, SalesReasonLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<SalesReasonLookupModel, BaseSalesReason>().IgnoreMissingDestinationMembers();
        }
    }
}
